swarm/src/dial_opts.rs

```diff
@@ -297,29 +297,34 @@ impl WithoutPeerIdWithAddress {
     }
 }
 
-/// The available conditions under which a new dialing attempt to
-/// a known peer is initiated.
-///
-/// ```
-/// # use libp2p_swarm::dial_opts::{DialOpts, PeerCondition};
-/// # use libp2p_identity::PeerId;
-/// #
-/// DialOpts::peer_id(PeerId::random())
-///    .condition(PeerCondition::Disconnected)
-///    .build();
-/// ```
-#[derive(Debug, Copy, Clone, Default)]
-pub enum PeerCondition {
-    /// A new dialing attempt is initiated _only if_ the peer is currently
-    /// considered disconnected, i.e. there is no established connection
-    /// and no ongoing dialing attempt.
-    #[default]
-    Disconnected,
-    /// A new dialing attempt is initiated _only if_ there is currently
-    /// no ongoing dialing attempt, i.e. the peer is either considered
-    /// disconnected or connected but without an ongoing dialing attempt.
-    NotDialing,
-    /// A new dialing attempt is always initiated, only subject to the
-    /// configured connection limits.
-    Always,
+bitflags::bitflags! {
+    /// The available conditions under which a new dialing attempt to
+    /// a known peer is initiated. These conditions can be combined.
+    ///
+    /// ```
+    /// # use libp2p_swarm::dial_opts::{DialOpts, PeerCondition};
+    /// # use libp2p_identity::PeerId;
+    /// #
+    /// DialOpts::peer_id(PeerId::random())
+    ///    .condition(PeerCondition::Disconnected | PeerCondition::NotDialing)
+    ///    .build();
+    /// ```
+    #[derive(Debug, Copy, Clone)]
+    pub struct PeerCondition: u32 {
+        /// A new dialing attempt is always initiated, only subject to the
+        /// configured connection limits.
+        const Always = 0b0000;
```

Comment:
```suggestion
        const Always = 0b0100;
```
 Shouldn't it be like that? Then `NotConnected & NotDialing` will be the same as `Always`.

On the other hand, no conditions (i.e. `empty`) should mean "always make the dial"). Perhaps an `Option` does make the most sense.

Reply:
With `Always = 0b0100`, it would be possible to combine e.g. `NotConnected | Always`. I wouldn't know what behavior to expect from that.

By the way, `NotConnected & NotDialing` would equal zero. Or did you mean `NotConnected | NotDialing`. Sorry, not trying to be pedantic, just making sure I understand what you're proposing.